docs/core/deploying/trimming/prepare-libraries-for-trimming.md

```diff
@@ -8,360 +8,234 @@ ms.date: 06/12/2023
 
-### [.NET 6](#tab/net6)
+The `IsTrimmable` property defaults to `true` when configuring a project as AOT-compatible with `<IsAotCompatible>true</IsAotCompatible>`. For more information, see [AOT-compatibility analyzers](../native-aot/index.md#aot-compatibility-analyzers).
 
-```xml
-<Project Sdk="Microsoft.NET.Sdk">
-
-  <PropertyGroup>
-    <OutputType>Exe</OutputType>
-    <!-- Replace with net5.0, or netcoreapp3.1 if testing for those frameworks -->
-    <TargetFramework>net6.0</TargetFramework>
-    <PublishTrimmed>true</PublishTrimmed>
-    <!-- Prevent warnings from unused code in dependencies -->
-    <TrimmerDefaultAction>link</TrimmerDefaultAction>
-  </PropertyGroup>
-
-  <ItemGroup>
-    <ProjectReference Include="path/to/MyLibrary.csproj" />
-    <!-- Analyze the whole library, even if attributed with "IsTrimmable" -->
-    <TrimmerRootAssembly Include="MyLibrary" />
-  </ItemGroup>
-
-</Project>
-```
+To generate trim warnings without marking the project as trim-compatible, use `<EnableTrimAnalyzer>true</EnableTrimAnalyzer>` rather than `<IsTrimmable>true</IsTrimmable>`.
 
-### [.NET 7](#tab/net7)
+### Show all warnings with test app
 
-```xml
-<Project Sdk="Microsoft.NET.Sdk">
+To show all analysis warnings for a library, the trimmer must analyze the implementation:
 
-  <PropertyGroup>
-    <OutputType>Exe</OutputType>
-    <TargetFramework>net7.0</TargetFramework>
-    <PublishTrimmed>true</PublishTrimmed>
-  </PropertyGroup>
+* Of the library.
+* All dependencies the library uses.
 
-  <ItemGroup>
-    <ProjectReference Include="path/to/MyLibrary.csproj" />
-    <!-- Analyze the whole library -->
-    <TrimmerRootAssembly Include="MyLibrary" />
-  </ItemGroup>
+When building and publishing a library:
 
-</Project>
-```
+* The implementations of the dependencies aren't available.
+* The available reference assemblies don't have enough information for the trimmer to determine if they're compatible with trimming.
 
-### [.NET 8+](#tab/net8plus)
+Because of the dependency limitations, a self-contained test app which uses the library and its dependencies must be created. The test app executable includes all the information the trimmer requires to issue warning on trim incompatibilities in:
 
-```xml
-<Project Sdk="Microsoft.NET.Sdk">
+* The library code.
+* The code that the library references from its dependencies.
 
-  <PropertyGroup>
-    <OutputType>Exe</OutputType>
-    <TargetFramework>net8.0</TargetFramework>
-    <PublishTrimmed>true</PublishTrimmed>
-  </PropertyGroup>
+***Note:*** If the library has different behavior depending on the target framework, create a trimming test app for each of the target frameworks. For example, if the library uses [conditional compilation](/dotnet/csharp/language-reference/preprocessor-directives#conditional-compilation) such as `#if NET7_0` to change behavior.
 
```

Comment:
This needs to take into account older TFMs where trimming is not possible. Lot of libraries will target `net472`, `netstandard2.0` and then for example `net6.0`. And they will often have ifdefs around these TFMs in the code. So as a library author I would read this as "I should do this" - while in reality, it will not work for anything but 6+.